controllers/kubeadmconfig_controller.go

```diff
@@ -449,30 +455,40 @@ func (r *KubeadmConfigReconciler) createClusterCertificates(ctx context.Context,
 		return nil, err
 	}
 
-	secret := &corev1.Secret{
-		ObjectMeta: v1.ObjectMeta{
-			Name:      ClusterCertificatesSecretName(clusterName),
-			Namespace: config.GetNamespace(),
-			OwnerReferences: []v1.OwnerReference{
-				{
-					APIVersion: cabpkv1alpha2.GroupVersion.String(),
-					Kind:       "KubeadmConfig",
-					Name:       config.GetName(),
-					UID:        config.GetUID(),
-				},
-			},
-		},
-		Data: certificates.ToMap(),
+	for _, secret := range certs.NewSecretsFromCertificates(certificates) {
+		linkOwnerReference(&secret.ObjectMeta, config)
+		secret.ObjectMeta.Labels["cluster"] = clusterName
+		secret.ObjectMeta.Name = prefixByClusterName(clusterName, secret.ObjectMeta.Name)
+		r.Log.Info("Creating secret for certificate", "name", secret.ObjectMeta.Name)
+		if err := r.Create(ctx, secret); err != nil {
+			return nil, err
+		}
 	}
 
-	if err := r.Create(ctx, secret); err != nil {
-		return nil, err
+	return certificates, nil
+}
+
+func linkOwnerReference(meta *v1.ObjectMeta, config *cabpkv1alpha2.KubeadmConfig) {
```

Comment:
I'd rather see this function generate and return the ownerReferences, and set the secret's ownerReferences in the for loop above using the output from this function.